criu/cr-restore.c

```diff
@@ -1438,7 +1438,9 @@ static inline int fork_with_pid(struct pstree_item *item)
 	}
 
 	if (ret < 0) {
-		pr_perror("Can't fork for %d", pid);
+		pr_perror("Can't fork for %d (errno %d)", pid, ret);
```

Comment:
Maybe "(errno %d)" is excess if you call pr_perror anyway? Except this looks good.

Reply:
I can remove that line. I just added it as pr_perror reports it as a string and the test case printed out that it expected 17 and comparing that to the errno string ('file exist') required an extra lookup for me.

Reply:
I'm not strictly against it but it attracts my eye as something strange. Let's leave it up to you and @avagin.

We already have 3 more places in code:

```
[snorch@snorch criu]$ git blame criu/cr-dedup.c | grep "pr_perror.*errno"
d866bd196b cr-dedup.c      (Tikhomirov Pavel 2013-12-17 19:27:08 +0400  22) 		pr_perror("Can't enter previous snapshot folder, error=%d", errno);
d866bd196b cr-dedup.c      (Tikhomirov Pavel 2013-12-17 19:27:08 +0400  32) 				pr_perror("Failed readdir, error=%d", errno);
[snorch@snorch criu]$ git blame criu/mem.c | grep "pr_perror.*errno"
ba78d15557 criu/mem.c (Kir Kolyshkin         2017-03-23 15:02:11 -0700   67) 			pr_perror("Can't reset %d's dirty memory tracker (%d)", pid, errno);
```

I'm guilty too =)

Reply:
We can add `errno %d` to `pr_error`, and drop these five cases.

Reply:
Or fix the tests. Maybe it's better to fix the tests.

Reply:
I removed the line, so we do not have to solve this discussion here.